From e127c30e51cf2b0ebf2a86ec6cf96a1992e7e22a Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 3 Jul 2025 16:48:45 +0300
Subject: [PATCH 5/6] Compulab: Integration of Mender boot code into U-Boot.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/compulab-imx8m-plus.h | 30 ++++++++++++++-------------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index 934a5be7e3..d2d5e48274 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -55,6 +55,7 @@
 	CFG_MFG_ENV_SETTINGS \
 	BOOTENV \
 	MACHINE_EXTRA_ENV_SETTINGS \
+	MENDER_ENV_SETTINGS \
 	"video_link=1\0" \
 	"stdout=serial,vidconsole\0" \
 	"stderr=serial,vidconsole\0" \
@@ -63,6 +64,7 @@
 	"scriptaddr=0x43500000\0" \
 	"kernel_addr_r=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"bsp_script=boot.scr\0" \
+	"bootdir=boot\0" \
 	"image=Image\0" \
 	"splashimage=0x50000000\0" \
 	SERIAL_CONSOLE "\0" \
@@ -78,12 +80,12 @@
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs console=${console} root=${mmcroot}\0 " \
-	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bsp_script};\0" \
+	"mmcargs=setenv bootargs console=${console} root=PARTUUID=${mender_uboot_root_name}\0 " \
+	"loadbootscript=load ${mender_uboot_root} ${loadaddr} ${bsp_script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr_r} ${fdtfile}\0" \
+	"loadimage=load ${mender_uboot_root} ${loadaddr} ${bootdir}/${image}\0" \
+	"loadfdt=load ${mender_uboot_root} ${fdt_addr_r} ${bootdir}/${fdtfile}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fit} = yes || test ${boot_fit} = try; then " \
@@ -105,11 +107,11 @@
 		"else " \
 			"setenv get_cmd tftp; " \
 		"fi; " \
-		"${get_cmd} ${loadaddr} ${image}; " \
+		"${get_cmd} ${loadaddr} ${bootdir}/${image}; " \
 		"if test ${boot_fit} = yes || test ${boot_fit} = try; then " \
 			"bootm ${loadaddr}; " \
 		"else " \
-			"if ${get_cmd} ${fdt_addr_r} ${fdtfile}; then " \
+			"if ${get_cmd} ${fdt_addr_r} ${bootdir}/${fdtfile}; then " \
 				"booti ${loadaddr} - ${fdt_addr_r}; " \
 			"else " \
 				"echo WARN: Cannot load the DT; " \
@@ -122,24 +124,24 @@
 		"boot_part=1\0" \
 		"root_opt=rootwait rw\0" \
 		"emmc_ul=setenv iface mmc; setenv dev ${emmc_dev}; setenv part ${boot_part};" \
-		"setenv bootargs console=${console} root=${emmc_root} ${root_opt};\0" \
+		"setenv bootargs console=${console} root=PARTUUID=${mender_uboot_root_name} ${root_opt};\0" \
 		"sd_ul=setenv iface mmc; setenv dev ${sd_dev}; setenv part ${boot_part};" \
-			"setenv bootargs console=${console} root=${sd_root} ${root_opt};\0" \
+			"setenv bootargs console=${console} root=PARTUUID=${mender_uboot_root_name} ${root_opt};\0" \
 		"usb_ul=usb start; setenv iface usb; setenv dev ${usb_dev}; setenv part ${boot_part};" \
-			"setenv bootargs console=${console} root=${usb_root} ${root_opt};\0" \
-		"ulbootscript=load ${iface} ${dev}:${part} ${scriptaddr} ${bsp_script};\0" \
-		"ulrunbootscript=echo Running bootscript from ${iface} ${dev}:${part} ...; " \
+			"setenv bootargs console=${console} root=PARTUUID=${mender_uboot_root_name} ${root_opt};\0" \
+		"ulbootscript=load ${mender_uboot_root} ${scriptaddr} ${bsp_script};\0" \
+		"ulrunbootscript=echo Running bootscript from ${mender_uboot_root} ...; " \
 				"source ${scriptaddr}\0" \
-		"ulimage=echo loading ${image}; load ${iface} ${dev}:${part} ${loadaddr} ${image}\0" \
+		"ulimage=echo loading ${bootdir}/${image}; load ${mender_uboot_root} ${loadaddr} ${bootdir}/${image}\0" \
 		"ulfdto=setenv fdto1file; for fdto1file in ${fdtofile}; do "\
 			    "echo loading ${fdto1file}; "\
-			    "load ${iface} ${dev}:${part} ${fdto_addr_r} ${fdto1file} "\
+			    "load ${mender_uboot_root} ${fdto_addr_r} ${fdto1file} "\
 			    "&& fdt addr ${fdt_addr_r} "\
 			    "&& fdt resize 0x8000 "\
 			    "&& fdt apply ${fdto_addr_r};"\
 		"done; true;\0"\
 		"ulfdt=if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-			"echo loading ${fdtfile}; load ${iface} ${dev}:${part} ${fdt_addr_r} ${fdtfile}; " \
+			"echo loading ${bootdir}/${fdtfile}; load ${mender_uboot_root} ${fdt_addr_r} ${bootdir}/${fdtfile}; " \
 			"if env exists fdtofile;then "\
 					"run ulfdto; "\
 				"else "\
-- 
2.34.1

