diff --git a/meta-mender-core/classes/mender-setup.bbclass b/meta-mender-core/classes/mender-setup.bbclass
index 574a904a..b0021734 100644
--- a/meta-mender-core/classes/mender-setup.bbclass
+++ b/meta-mender-core/classes/mender-setup.bbclass
@@ -331,7 +331,7 @@ python mender_sanity_handler() {
     if bb.utils.contains('MENDER_FEATURES_ENABLE', 'mender-partuuid', True, False, d) and d.getVar('MENDER_STORAGE_DEVICE', True) != "":
         bb.warn("MENDER_STORAGE_DEVICE is ignored when mender-partuuid is enabled. Clear MENDER_STORAGE_DEVICE to remove this warning.")
 
-    if bb.utils.contains('MENDER_FEATURES_ENABLE', 'mender-partuuid', True, False, d) and not bb.utils.contains_any('TUNE_ARCH', 'i586 i686 x86 x86_64', True, False, d):
+    if bb.utils.contains('MENDER_FEATURES_ENABLE', 'mender-partuuid', True, False, d) and not bb.utils.contains_any('TUNE_ARCH', 'i586 i686 x86 x86_64 arm', True, False, d):
         bb.fatal("mender-partuuid is not supported on the %s architecture." % d.getVar("TUNE_ARCH"))
 }
 
